File: Examples.vb/ConversionExamples/DoubleTryParse.vb
Option Explicit On

Imports NUnit.Framework

Namespace ConversionExamples
	<TestFixture>
	Public Class DoubleTryParse

		<Test> <SetCulture("en-US")>
		Public Sub DoubleTryParseExample_enUS()

			Assert.That(Double.TryParse("1,5", Nothing), [Is].True)
			Assert.That(Double.TryParse("1.5", Nothing), [Is].True)

		End Sub

		<Test> <SetCulture("sv-SE")>
		Public Sub DoubleTryParseExample_svSE()

			Assert.That(Double.TryParse("1,5", Nothing), [Is].True)
			Assert.That(Double.TryParse("1.5", Nothing), [Is].False)

		End Sub

	End Class
End Namespace

File: Nunit.Test/Examples/TryParse/DoubleTryParse.cs
using NUnit.Framework;
using System;

namespace Examples.TryParse
{
	[TestFixture]
	public class DoubleTryParse
	{
		[Test]
		[SetCulture("en-US")]
		public void DoubeTryParseExample_enUS()
		{
			double testVar;
			Assert.That(Double.TryParse("1,5", out testVar), Is.True);
			Assert.That(Double.TryParse("1.5", out testVar), Is.True);
		}

		[Test]
		[SetCulture("en-UK")]
		public void DoubeTryParseExample_enUK()
		{
			double testVar;
			Assert.That(Double.TryParse("1,5", out testVar), Is.True);
			Assert.That(Double.TryParse("1.5", out testVar), Is.True);
		}

		[Test]
		[SetCulture("sv-SE")]
		public void DoubeTryParseExample_svSE()
		{
			double testVar;
			Assert.That(Double.TryParse("1,5", out testVar), Is.True);
			Assert.That(Double.TryParse("1.5", out testVar), Is.False);
		}
	}
}
